client/src/components/checkout-form.js

```diff
@@ -0,0 +1,158 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { LitElement, html } from 'lit';
+import styles from './styles/checkout-form.js';
+
+class CheckoutForm extends LitElement {
+  static properties() {
+    return {
+      onSubmit: { type: Function },
+    };
+  }
+
+  static get styles() {
+    return styles;
+  }
+
+  constructor() {
+    super();
+    this.onSubmit = () => {};
+    this.submitForm = this.submitForm.bind(this);
+
+    this.state = {
+      openFormErrorDialog: false,
+      openSuccessDialog: false,
+      openErrorDialog: false,
+    };
+  }
+
+  toggleFormErrorDialog() {
+    this.state.openFormErrorDialog = !this.state.openFormErrorDialog;
+    this.requestUpdate();
+  }
+
+  toggleSuccessDialog() {
+    this.state.openSuccessDialog = !this.state.openSuccessDialog;
+    this.requestUpdate();
+  }
+
+  toggleErrorDialog() {
+    this.state.openErrorDialog = !this.state.openErrorDialog;
+    this.requestUpdate();
+  }
+
+  async submitForm(event) {
+    event?.preventDefault();
+    const form = new FormData(this.shadowRoot.querySelector('form') || {});
+    let response;
+
+    if (this.isValidEmail(form.get('email'))) {
+      try {
+        response = await this.onSubmit(form);
+
+        // TODO: getting called to earlier, moving up to parent checkout page to handle success/error handling
+        response?.status?.ok
```

Comment:
I'm also seeing that there is no `response.status.ok`, only a `response.ok`, so suggest: 
```suggestion
        response?.ok
```

In my local test, if I select 'collect', and have a valid email, and don't ask for too many items, I can get a successful cart transaction popup

Reply:
Moving this up to parent component! Yup, that was an oversight! 🤦🏼‍♀️ Ty for catching!